### PR TITLE
[Merged by Bors] - chore: split FieldTheory/Tower 

### PR DESCRIPTION
**Pull request by kim-em**
The results in this file were unrelated (and had disjoint import requirements!) so I've separated them.

The doc-string was badly out of date before I came along, so I've replaced it.

### CODE REVIEW

**Comment by semorrison** [comment]
Oops, thanks @Parcly-Taxel!!

File: Mathlib/Algebra/Algebra/Subalgebra/IsSimpleOrder.lean [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import Mathlib.LinearAlgebra.FreeModule.StrongRankCondition
+
+/-!
+If `A` is a domain, and a finite-dimensional algebra over a field `F`, with prime dimension,
+then there are no non-trivial `F`-subalgebras.
+-/
+
+open FiniteDimensional Submodule
+
+theorem Subalgebra.isSimpleOrder_of_finrank_prime (F A) [Field F] [Ring A] [IsDomain A]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Algebra/Subalgebra/IsSimpleOrder.lean:17]: I think you renamed this from `FiniteDimensional.Subalgebra.isSimpleOrder_of_finrank_prime`

Comment (semorrison) [reply Mathlib/Algebra/Algebra/Subalgebra/IsSimpleOrder.lean:17]: I'm pretty sure that was a mistake? I didn't add a deprecation as it is unused in Mathlib.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**